src/status_im/ui/screens/discover/events.cljs

```diff
@@ -1,148 +1,160 @@
 (ns status-im.ui.screens.discover.events
-  (:require [re-frame.core :refer [after dispatch enrich]]
-            [status-im.utils.utils :refer [first-index]]
-            [status-im.utils.handlers :refer [register-handler get-hashtags]]
+  (:require [re-frame.core :as re-frame]
             [status-im.protocol.core :as protocol]
-            [status-im.ui.screens.navigation :as nav]
-            [status-im.data-store.discover :as discoveries]
-            [status-im.utils.handlers :as u]
-            [status-im.utils.datetime :as time]
-            [status-im.utils.random :as random]
-            [taoensso.timbre :as log]
+            [status-im.ui.screens.discover.navigation]
             [status-im.utils.handlers :as handlers]))
 
 (def request-discoveries-interval-s 600)
+(def maximum-number-of-discoveries 1000)
 
-(register-handler :init-discoveries
-  (fn [db _]
-    (-> db
-        (assoc :tags [])
-        (assoc :discoveries {}))))
-
-(defmethod nav/preload-data! :discover
-  [db _]
-  (-> db
-      (assoc-in [:toolbar-search :show] nil)
-      (assoc :tags (discoveries/get-all-tags))
-      (assoc :discoveries (->> (discoveries/get-all :desc)
-                               (map (fn [{:keys [message-id] :as discover}]
-                                      [message-id discover]))
-                               (into {})))))
-
-;; todo(goranjovic): at the moment we do nothing when a status without hashtags is posted
+;; EFFECTS
+
+(re-frame/reg-fx
+  ::send-portions
+  (fn [{:keys [current-public-key web3 contacts to discoveries]}]
+    (protocol/send-discoveries-response!
```

Comment:
Shouldn't we create effects mapped directly to the `protocol` methods and use them ? It would facilitate better re-use and less logic in effects.

Reply:
yes probably, but I would wait to see how @flexsurfer refactors the protocols before doing such things.